ibis/tests/expr/test_table.py

```diff
@@ -2077,3 +2077,87 @@ def test_unbind_with_namespace():
 
     assert s.op() == expected.op()
     assert s.equals(expected)
+
+
+def test_table_bind():
+    def eq(left, right):
+        return all(a.equals(b) for a, b in zip(left, right))
+
+    t = ibis.table({"a": "int", "b": "string"}, name="t")
+
+    # boolean literals
+    exprs = t.bind(True, False)
+    expected = (ibis.literal(True), ibis.literal(False))
+    assert eq(exprs, expected)
+
```

Comment:
Would love 2 cases, one for for a completely unrelated table `ibis.table({"z": "int"}).bind(t)` and another for a related one (I can only think of a join result but maybe there's some other instance of this) `t.right_join(ibis.table({"a": "int"}), "a").bind(t.b)`. I'm not that attached to what the behavior is, but I think we should write it down so we can test for future breakages.

Reply:
Don't we already have a test for the latter?

Reply:
do you mean the `t.bind(t)` case? I was hoping for something a bit more complex. Or did I miss something?

Reply:
I think it is a blocker for me, at the very least to simply raise a NotImplemented error so we keep the door open for ourselves later

Reply:
oooh, actually there is also `t.bind(unrelated_table.column)` which currently returns `unrelated_table.column`, which I don't think quite makes sense (I would expect an error unless we can walk up the join chain to find it). So actually all this seems like undefined behavior. I think if we just don't publicize this API or add it to the type hints then I'm happy. Would love to make this more flushed out later but it is a bigger task than we should block on.

Reply:
Or I guess per the conversation below, maybe again this isn't the responsibility of .bind to do this check